rfcs/text/0004-session.md

```diff
@@ -88,62 +38,85 @@ Stage 3: Add or update all remaining field definitions. The list should now be e
 
 ## Usage
 
-Session fields are used to describe the sesison attributes of:
- - Client VPN Sessions
- - Network to Network VPN Sessions
- - Network Access Sessions (NAC, WPA, EAP, etc.)
- - Local or remote device login sessions (RDP, ICA, xWindows)
+Session fields are used to describe and track a discrete grouping of interactions, typically bounded by
+authentication or authorization events and tied to a specific user, application, or system component.
+
+For example:
+ - Network Access Sessions (NAC or Wireless LAN)
+ - Local or remote device login sessions (tty, console, ssh, RDP, ICA, xWindows, etc.)
+ - VPN Sessions (network to network, or client to network)
+ - Local or remote device login sessions (console, tty, RDP, ICA, xWindows, ssh, etc.)
  - Administrative sessions on infrastructure devices
  - Administrative sessions on cloud or application management portals
- - Applications sessions (e.g. sql server odbc session, application access session)
+ - Applications sessions (e.g. sql server odbc session, web cookie based session, application access session)
+ - Cloud API access sessions
 
 
 ## Source data
 Source data expectations include:
  - Wireless Lan Controllers
- - Security appliances
+ - Security appliances (e.g. fw, waf)
  - Network admission control devices
- - Radius / tacacs servers
- - Application server logs
+ - Radius / tacacs servers (802.1x EAP/PEAP aaa)
+ - Application server logs (FTP, MySQL)
+ - Web Server, WAF, or ADC logs (USer or cookie based web ession control)
+ - APM telemetry
```

Comment:
For our RUM agent, running on client websites tracking browsing sessions,  I can only see us setting `session.kind` and `session.id`. Would love feedback from the RFC authors on whether this makes sense.

CC @jahtalab

Reply:
I think the idea (only filling out relevant fields) makes total sense... tho in respect to this and the previous comment it makes me think I'm missing something that identifies a RUM type session (but then I also don't really know what other fields would be coming in that would make filtering results from a mixed index easy). Unauthenticated web sessions for instance would't need to use session.auth, but after a login could (along with user.name, etc.). I don't see that one as big a deal as e.g. network access logins (vpn, wireless, etc)

Reply:
To add some context, the RUM sessions are not necessarily authenticated, even until they are complete, e.g. a user session for going through checkout.

Reply:
@jahtalab do we have any notion of auth no? AFAIK we don't, and it's unlikely given the extreme variety of auth methods.

Reply:
@andrewvc , We don't detect authentication per se, but there's a configuration for customers to add the user (username, id, ..) in the RUM agent. We can assume the user is authenticated (with some method) but it doesn't have to be.